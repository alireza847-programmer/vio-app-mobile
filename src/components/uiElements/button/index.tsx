import React from 'react';
import Button from './style';
import {VButtonProps} from 'types/components/uiElements/button';
import VText from '../text';
import {theme} from 'themes/emotion';
import VRow from '../row';

const VButton = (props: VButtonProps) => {
  const {
    mode = 'primary',
    subTitle,
    title,
    styled = 'FULL_WIDTH',
    icon,
    rightIcon,
    onPress,
    testID,
    disabled = false,
    width,
  } = props;

  const renderContentColor = () => {
    if (disabled) {
      return theme.colors.textColor.button.disabled;
    }
    if (styled === 'TEXT') {
      return theme.colors.button[mode];
    }
    return theme.colors.textColor.button[mode];
  };

  return (
    <Button
      width={width}
      disabled={disabled}
      testID={testID}
      activeOpacity={0.8}
      onPress={onPress}
      styled={styled}
      mode={mode}>
      {icon && (
        <VRow {...(title && {marginRighRatio: 2})} fullWidth={false}>
          {icon(renderContentColor())}
        </VRow>
      )}
      <VText
        typography="semiBold16"
        color={renderContentColor()}
        marginTopRatio={0}>
        {title}
      </VText>
      {subTitle && (
        <VText
          typography="regular16"
          color={renderContentColor()}
          marginLeftRatio={1.2}
          marginTopRatio={0}>
          {subTitle}
        </VText>
      )}
      {rightIcon && (
        <VRow {...(title && {marginLeftRatio: 2})} fullWidth={false}>
          {rightIcon(renderContentColor())}
        </VRow>
      )}
    </Button>
  );
};
export default VButton;
