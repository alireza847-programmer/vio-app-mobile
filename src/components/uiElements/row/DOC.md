# VRow Component Documentation

## Overview

The `VRow` component is a versatile row container designed for use in React Native projects. It provides flexibility in styling, allowing you to easily manage the layout of child components within a row.

## Usage

```jsx
import VRow, {RowProps} from 'path-to-your-component/VRow';

const MyComponent = () => {
  return (
    <VRow
      justifyContent="flex-start"
      alignItems="center"
      backgroundColor="#f0f0f0"
      fullWidth={true}
      marginTopRatio={2}
      marginLeftRatio={1}
      paddingHorizontalRatio={1}
      paddingVerticalRatio={1}>
      {/* Your child components here */}
    </VRow>
  );
};
```

## Props

justifyContent (optional): Control the horizontal alignment of child components. Options include 'center', 'flex-start', 'flex-end', 'space-between', 'space-around', and 'space-evenly'.

alignItems (required): Set the vertical alignment of child components. Options include 'flex-start', 'flex-end', 'center', 'stretch', and 'baseline'.

backgroundColor (optional): Apply a background color to the row container.

fullWidth (optional): If set to true, the row will take up the full width of its parent container. Default is true.

marginTopRatio (optional): Adjust the top margin of the row based on a ratio.

marginLeftRatio (optional): Adjust the left margin of the row based on a ratio.

paddingHorizontalRatio (optional): Adjust the horizontal padding of the row based on a ratio.

paddingVerticalRatio (optional): Adjust the vertical padding of the row based on a ratio.
