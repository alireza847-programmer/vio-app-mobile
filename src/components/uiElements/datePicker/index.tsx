import {theme} from '@storybook/react-native';
import React from 'react';
import {View} from 'react-native';
import Calendar from 'react-native-calendar-range-picker';
import {VDatePickerProps} from 'types/components/uiElements/datePicker';

const VDatePicker = (props: VDatePickerProps) => {
  const {...rest} = props;
  return (
    <View style={{width: '100%'}}>
      <Calendar disabledBeforeToday pastYearRange={0} {...rest} />
    </View>
  );
};

export default VDatePicker;
