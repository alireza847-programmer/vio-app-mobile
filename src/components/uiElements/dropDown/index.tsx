import React, {Fragment, useEffect, useState} from 'react';
import {VDropDownProps} from 'types/components/uiElements/dropDown';
import {theme} from 'themes/emotion';
import {ChevronSvg} from 'assets/svgs';
import {Container} from './style';
import WheelPicker from 'react-native-wheely';
import VModal from '../modal';
import VText from '../text';
import VRow from '../row';
import VButton from '../button';
import {TouchableOpacity} from 'react-native';

const VDropDown = (props: VDropDownProps) => {
  const {data, placeHolder, onChange, defaultValue} = props;
  const [visibleModal, setVisibleModal] = useState(false);
  const [value, setValue] = useState('');
  const [selectedIndex, setSelectedIndex] = useState(0);
  useEffect(() => {
    if (defaultValue) {
      setValue(String(defaultValue));
      setSelectedIndex(data.indexOf(String(defaultValue)));
    } else {
      setValue(String(placeHolder));
    }
  }, [placeHolder, defaultValue]);
  const onConfirm = () => {
    setVisibleModal(false);
    setValue(data[selectedIndex]);
    onChange(data[selectedIndex]);
  };
  const renderModal = () => {
    return (
      <VModal
        withClose
        withHeader
        title="Child 1 age"
        onClose={() => setVisibleModal(false)}
        isVisible={visibleModal}>
        <WheelPicker
          selectedIndex={selectedIndex}
          options={data}
          onChange={index => setSelectedIndex(index)}
          containerStyle={{width: '90%'}}
          opacityFunction={x => 0.3}
          selectedIndicatorStyle={{
            backgroundColor: theme.colors.button.secondary,
            borderTopWidth: 1,
            borderBottomWidth: 1,
            borderColor: theme.colors.button.primary,
          }}
          visibleRest={1}
          itemStyle={{
            backgroundColor: 'transparent',
          }}
          itemTextStyle={{
            ...theme.fonts.semiBold18,
          }}
        />
        <VRow paddingVerticalRatio={4} justifyContent="center">
          <VButton onPress={onConfirm} width={'90%'} title="CONFIRM" />
        </VRow>
      </VModal>
    );
  };
  return (
    <Fragment>
      <TouchableOpacity
        activeOpacity={0.8}
        onPress={() => setVisibleModal(true)}>
        <Container
          alignItems="center"
          justifyContent="space-between"
          fullWidth={false}>
          <VText
            width={theme.HorizontalRem * 10}
            textAlign="left"
            typography="semiBold16">
            {value}
          </VText>
          <VRow fullWidth={false} marginLeftRatio={3}>
            <ChevronSvg fill={theme.colors.button.primary} />
          </VRow>
        </Container>
      </TouchableOpacity>
      {renderModal()}
    </Fragment>
  );
};
export default VDropDown;
