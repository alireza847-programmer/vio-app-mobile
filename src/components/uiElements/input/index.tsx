import React from 'react';
import Input from './style';
import {VInputProps} from 'types/components/uiElements/input';
import {theme} from 'themes/emotion';

const VInput = (props: VInputProps) => {
  const {...rest} = props;
  return (
    <Input
      placeholderTextColor={theme.colors.textColor.placeHolder}
      {...rest}
    />
  );
};

export default VInput;
