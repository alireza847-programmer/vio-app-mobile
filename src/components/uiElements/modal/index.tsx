import Modal from 'react-native-modal';
import React, {PropsWithChildren} from 'react';
import {VModalProps} from 'types/components/uiElements/modal';
import {HeaderContainer, Wrapper, modalStyle} from './style';
import {theme} from 'themes/emotion';
import {CloseSvg} from 'assets/svgs';
import VText from '../text';
import VButton from '../button';

const VModal = (props: PropsWithChildren<VModalProps>) => {
  const {children, isVisible, onClose, title, withClose, withHeader} = props;
  return (
    <Modal
      animationIn="slideInUp"
      animationOut={'slideOutDown'}
      style={modalStyle}
      testID="modal"
      hasBackdrop={true}
      onBackButtonPress={onClose}
      onBackdropPress={() => onClose()}
      isVisible={isVisible}>
      <Wrapper>
        {withHeader && (
          <HeaderContainer
            justifyContent="space-between"
            paddingHorizontalRatio={4}
            paddingVerticalRatio={6}>
            {withClose && (
              <VButton
                onPress={onClose}
                testID="close-button"
                styled="TEXT"
                icon={fill => <CloseSvg fill={theme.colors.button.primary} />}
              />
            )}
            <VText typography="semiBold16">{title}</VText>
            <CloseSvg fill={'transparent'} />
          </HeaderContainer>
        )}
        {children}
      </Wrapper>
    </Modal>
  );
};
export default VModal;
