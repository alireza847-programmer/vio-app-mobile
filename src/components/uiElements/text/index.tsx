import styled from '@emotion/native';
import {TextProps} from 'types/components/uiElements/text';

const VText = styled.Text<TextProps>(
  ({
    theme,
    textAlign = 'center',
    typography = 'medium15',
    color = theme.colors.textColor.primary,
    marginTopRatio = 0,
    marginLeftRatio = 0,
    fullWidth = false,
    width,
  }) => ({
    color: color,
    textAlign: textAlign,
    marginTop: theme.verticalRem * marginTopRatio,
    marginLeft: theme.verticalRem * marginLeftRatio,
    textAlignVertical: 'center',
    ...(fullWidth && {width: '100%'}),
    ...theme.fonts[typography],
    ...(width && {width: width}),
  }),
);

export default VText;
