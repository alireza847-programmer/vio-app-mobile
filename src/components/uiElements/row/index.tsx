import styled from '@emotion/native';
import {PropsWithChildren} from 'react';
import {RowProps} from 'types/components/uiElements/row';

const VRow = styled.View<PropsWithChildren<RowProps>>(
  ({
    theme,
    backgroundColor,
    fullWidth = true,
    justifyContent = 'flex-start',
    alignItems = 'center',
    marginTopRatio = 0,
    marginLeftRatio = 0,
    paddingHorizontalRatio = 0,
    paddingVerticalRatio = 0,
    marginRighRatio = 0,
    minWidth,
  }) => ({
    flexDirection: 'row',
    justifyContent,
    alignItems,
    marginTop: theme.verticalRem * marginTopRatio,
    marginLeft: theme.HorizontalRem * marginLeftRatio,
    marginRight: theme.HorizontalRem * marginRighRatio,
    paddingHorizontal: theme.HorizontalRem * paddingHorizontalRatio,
    paddingVertical: theme.verticalRem * paddingVerticalRatio,
    ...(backgroundColor && {backgroundColor}),
    ...(fullWidth && {width: '100%'}),
    ...(minWidth && {minWidth}),
  }),
);

export default VRow;
