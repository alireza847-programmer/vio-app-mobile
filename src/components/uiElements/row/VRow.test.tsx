import React, {PropsWithChildren} from 'react';
import {render} from '@testing-library/react-native';
import VRow from './index';
import {it, describe, expect, jest} from '@jest/globals';
import {ThemeProvider} from '@emotion/react';
import {theme} from 'themes/emotion';
import {VRowProps} from 'types/components/uiElements/row';

const Component = (props: PropsWithChildren<VRowProps>) => (
  <ThemeProvider theme={theme}>
    <VRow {...props}>{props.children}</VRow>
  </ThemeProvider>
);

describe('VRow component', () => {
  it('matches snapshot with default props', () => {
    const {toJSON} = render(<VRow>Content goes here</VRow>);
    expect(toJSON()).toMatchSnapshot();
  });

  it('matches snapshot with custom background color', () => {
    const {toJSON} = render(
      <Component backgroundColor="blue">Blue Background</Component>,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('matches snapshot with custom styling and justifyContent', () => {
    const {toJSON} = render(
      <Component
        justifyContent="space-between"
        marginTopRatio={1}
        marginLeftRatio={2}>
        Custom Styling
      </Component>,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('matches snapshot with fullWidth prop', () => {
    const {toJSON} = render(<Component fullWidth>Full Width Row</Component>);
    expect(toJSON()).toMatchSnapshot();
  });

  it('matches snapshot with custom minWidth', () => {
    const {toJSON} = render(
      <Component minWidth={150}>Min Width Row</Component>,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
