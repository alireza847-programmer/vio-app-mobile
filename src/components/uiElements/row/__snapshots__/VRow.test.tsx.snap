// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VRow component matches snapshot with custom background color 1`] = `
<View
  backgroundColor="blue"
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "blue",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 0,
        "paddingHorizontal": 0,
        "paddingVertical": 0,
        "width": "100%",
      },
      undefined,
    ]
  }
>
  Blue Background
</View>
`;

exports[`VRow component matches snapshot with custom minWidth 1`] = `
<View
  minWidth={150}
  style={
    [
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 0,
        "minWidth": 150,
        "paddingHorizontal": 0,
        "paddingVertical": 0,
        "width": "100%",
      },
      undefined,
    ]
  }
>
  Min Width Row
</View>
`;

exports[`VRow component matches snapshot with custom styling and justifyContent 1`] = `
<View
  justifyContent="space-between"
  marginLeftRatio={2}
  marginTopRatio={1}
  style={
    [
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginLeft": 15,
        "marginRight": 0,
        "marginTop": 6.67,
        "paddingHorizontal": 0,
        "paddingVertical": 0,
        "width": "100%",
      },
      undefined,
    ]
  }
>
  Custom Styling
</View>
`;

exports[`VRow component matches snapshot with default props 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "marginLeft": NaN,
        "marginRight": NaN,
        "marginTop": NaN,
        "paddingHorizontal": NaN,
        "paddingVertical": NaN,
        "width": "100%",
      },
      undefined,
    ]
  }
>
  Content goes here
</View>
`;

exports[`VRow component matches snapshot with fullWidth prop 1`] = `
<View
  fullWidth={true}
  style={
    [
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 0,
        "paddingHorizontal": 0,
        "paddingVertical": 0,
        "width": "100%",
      },
      undefined,
    ]
  }
>
  Full Width Row
</View>
`;

exports[`VRow component should match snapshot with custom background color 1`] = `
<View
  backgroundColor="blue"
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "blue",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 0,
        "paddingHorizontal": 0,
        "paddingVertical": 0,
        "width": "100%",
      },
      undefined,
    ]
  }
>
  Blue Background
</View>
`;

exports[`VRow component should match snapshot with custom minWidth 1`] = `
<View
  minWidth={150}
  style={
    [
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 0,
        "minWidth": 150,
        "paddingHorizontal": 0,
        "paddingVertical": 0,
        "width": "100%",
      },
      undefined,
    ]
  }
>
  Min Width Row
</View>
`;

exports[`VRow component should match snapshot with custom styling and justifyContent 1`] = `
<View
  justifyContent="space-between"
  marginLeftRatio={2}
  marginTopRatio={1}
  style={
    [
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginLeft": 15,
        "marginRight": 0,
        "marginTop": 6.67,
        "paddingHorizontal": 0,
        "paddingVertical": 0,
        "width": "100%",
      },
      undefined,
    ]
  }
>
  Custom Styling
</View>
`;

exports[`VRow component should match snapshot with default props 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "marginLeft": NaN,
        "marginRight": NaN,
        "marginTop": NaN,
        "paddingHorizontal": NaN,
        "paddingVertical": NaN,
        "width": "100%",
      },
      undefined,
    ]
  }
>
  Content goes here
</View>
`;

exports[`VRow component should match snapshot with fullWidth prop 1`] = `
<View
  fullWidth={true}
  style={
    [
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 0,
        "paddingHorizontal": 0,
        "paddingVertical": 0,
        "width": "100%",
      },
      undefined,
    ]
  }
>
  Full Width Row
</View>
`;
